### ISSUE TITLE
imaplib.abort: socket error: EOF

### ISSUE THREAD

**Issue by teranex** [isbg/isbg#37]
I run isbg every 4 minutes on my RaspberryPI to check e-mail on my dreamhost account. Most of the time it works without any problem, but for some mails which spamassassin marks as spam isbg crashes with the below exception. Next time isbg will crash again and again, until I manually remove that specific spam-mail from my inbox:

```
Traceback (most recent call last):
  File "/opt/isbg/isbg.py", line 591, in <module>
    res = imap.append(spaminbox, None, None, body)
  File "/usr/lib/python2.7/imaplib.py", line 324, in append
    return self._simple_command(name, mailbox, flags, date_time)
  File "/usr/lib/python2.7/imaplib.py", line 1070, in _simple_command
    return self._command_complete(name, self._command(name, *args))
  File "/usr/lib/python2.7/imaplib.py", line 867, in _command
    while self._get_response():
  File "/usr/lib/python2.7/imaplib.py", line 916, in _get_response
    resp = self._get_line()
  File "/usr/lib/python2.7/imaplib.py", line 1011, in _get_line
    raise self.abort('socket error: EOF')
imaplib.abort: socket error: EOF
```

After reading [this stackoverflow page](http://stackoverflow.com/questions/7575943/eof-error-in-imaplib), it seems I was able to hack around this problem  by changing line 591 from this:

``` python
res = imap.append(spaminbox, None, None, body)
```

into this (obviously not a pretty hack for general use):

``` python
try:    
  res = imap.append(spaminbox, None, None, body)                                                                                
except imap.abort:
  imap = imaplib.IMAP4_SSL(imaphost, imapport)
  imap.login(imapuser, imappasswd)
  imap.select(imapinbox, 1)
  res = imap.append(spaminbox, None, None, body)
```

**Comment by baldurmen** [comment]
Woot, thanks for the bug report. Bug reports are always warmly welcomed.

I'm curious: is the email where it hangs special in any way?

This looks like a imaplib error though. If I were you, I'd report it on their bug tracker. You should not have to reconnect to be able to read the message...

Can you check on the python bug tracker ([here](https://bugs.python.org/issue?%40columns=id%2Cactivity%2Ctitle%2Ccreator%2Cassignee%2Cstatus%2Ctype&%40sort=-activity&%40filter=status&%40action=searchid&ignore=file%3Acontent&%40search_text=imaplib&submit=search&status=-1%2C1%2C2%2C3)) to see if there is already an opened issue? Creating one if it is not the case would be very nice too.

**Comment by dc55028** [comment]
same issue here.
running in a docker container on a RPi2 (domcomte/imapscan-rpi2), worked fine for nearly 2 months and then the network errors became more and more frequent and it seems unrelated to the age of the container.

**Comment by baldurmen** [comment]
This is fixed by #86 (it loops the login request 10 times until it works).
